cmd/kubeadm/app/constants/constants.go

```diff
@@ -399,15 +402,29 @@ const (
 )
 
 var (
+	// MasterTaint is the taint to apply on the PodSpec for being able to run that Pod on the control-plane
+	// DEPRECATED: https://github.com/kubernetes/kubeadm/issues/2200
+	MasterTaint = v1.Taint{
+		Key:    LabelNodeRoleMaster,
+		Effect: v1.TaintEffectNoSchedule,
+	}
+
+	// MasterToleration is the toleration to apply on the PodSpec for being able to run that Pod on the control-plane
+	// DEPRECATED: https://github.com/kubernetes/kubeadm/issues/2200
+	MasterToleration = v1.Toleration{
+		Key:    LabelNodeRoleMaster,
+		Effect: v1.TaintEffectNoSchedule,
+	}
+
 	// ControlPlaneTaint is the taint to apply on the PodSpec for being able to run that Pod on the control-plane
 	ControlPlaneTaint = v1.Taint{
-		Key:    LabelNodeRoleMaster,
+		Key:    LabelNodeRoleControlPlane,
```

Comment:
@neolit123 - If someone rollbacks would the node object still have this taint ? (thus preventing potential user services running the control plane from scheduling) ?

Reply:
In this pr we are not yet adding the taint to node objects. But your cobcern is valid when we start doind that. Potentially if kubeadm upgrade fails, we could have logic to remove the taint near the step of restoring the backed up manifests.